### PR TITLE
Update protocols

### PR DESCRIPTION
**Pull request by jburel**
Drop 1, 1.1 and add support for 1.3
Also add ciphers

### CODE REVIEW

**Comment by pwalczysko** [comment]
I have run this change against our www VM as per https://github.com/openmicroscopy/management_tools/pull/1768 with the nginx_proxy role pointing to this branch.

In ``--dry-run`` I have as a change


```
TASK [nginx_proxy : nginx | proxy config] **************************************
--- before: /etc/nginx/conf.d/proxy-default.conf
+++ after: /Users/pwalczysko/.ansible/tmp/ansible-local-49811bneju5zk/tmpsrno9khk/nginx-confd-proxy.j2
@@ -12,7 +12,9 @@
 
     ssl_certificate /etc/ssl/localcerts/bundled.crt;
     ssl_certificate_key /etc/ssl/localcerts/server.key;
-    ssl_protocols  TLSv1 TLSv1.1 TLSv1.2;  # don't use SSLv3 ref: POODLE
+    ssl_protocols  TLSv1.2 TLSv1.3;  # don't use SSLv3 ref: POODLE
+    ssl_ciphers: ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:DHE-RSA-CHACHA20-POLY1305;
+
 
     #charset koi8-r;
     #access_log  /var/log/nginx/log/host.access.log  main;

changed: [134.36.6.86] => (item={'nginx_proxy_is_default': True})

```

@jburel happy for me to run it in wet run ?

**Comment by jburel** [comment]
That's what I will expect. The change has manually been done on www yesterday
This is for the website on new VM

**Comment by pwalczysko** [comment]
Sorry, on wet run we have a failure of nginx to start

```
Starting The nginx HTTP and reverse proxy server...
: nginx: [emerg] unknown directive "ssl_ciphers:" in /etc/nginx/conf.d/proxy-default.conf:16
```

We will have to revisit this.

**Comment by pwalczysko** [comment]
the problem is the colon after the item

``ssl_ciphers:``

When colon ^^^ is removed, nginx restarts

**Comment by jburel** [comment]
Well spotted

File: templates/nginx-confd-proxy.j2 [hunk 1]
```diff
@@ -35,7 +35,9 @@ server {
 {% if (site.nginx_proxy_ssl | default(nginx_proxy_ssl)) %}
     ssl_certificate {{ site.nginx_proxy_ssl_certificate | default(nginx_proxy_ssl_certificate) }};
     ssl_certificate_key {{ site.nginx_proxy_ssl_certificate_key | default(nginx_proxy_ssl_certificate_key) }};
-    ssl_protocols  TLSv1 TLSv1.1 TLSv1.2;  # don't use SSLv3 ref: POODLE
+    ssl_protocols  TLSv1.2 TLSv1.3;  # don't use SSLv3 ref: POODLE
+    ssl_ciphers: ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:DHE-RSA-CHACHA20-POLY1305;
```
Comment (pwalczysko) [inline templates/nginx-confd-proxy.j2:39]: remove the colon in ``ssl_ciphers:`` as it fails the nginx restart

**Review by pwalczysko — CHANGES_REQUESTED** [review]
Please remove colon https://github.com/ome/ansible-role-nginx-proxy/pull/40/files#r1850294748

**Comment by jburel** [comment]
@pwalczysko done

**Review by sbesson — APPROVED** [review]
Makes complete sense to have this role allowing TLS 1.2 or higher. Should we target to get this deployed in `test125` then `prod125`?

**Review by pwalczysko — APPROVED** [review]
wet rerun went fine with

```
TASK [nginx_proxy : nginx | proxy config] **************************************
--- before: /etc/nginx/conf.d/proxy-default.conf
+++ after: /Users/pwalczysko/.ansible/tmp/ansible-local-63229cs12nzh2/tmplaz28jss/nginx-confd-proxy.j2
@@ -12,7 +12,9 @@
 
     ssl_certificate /etc/ssl/localcerts/bundled.crt;
     ssl_certificate_key /etc/ssl/localcerts/server.key;
-    ssl_protocols  TLSv1 TLSv1.1 TLSv1.2;  # don't use SSLv3 ref: POODLE
+    ssl_protocols  TLSv1.2 TLSv1.3;  # don't use SSLv3 ref: POODLE
+    ssl_ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:DHE-RSA-CHACHA20-POLY1305;
+
 
     #charset koi8-r;
     #access_log  /var/log/nginx/log/host.access.log  main;

changed: [134.36.6.86] => (item={'nginx_proxy_is_default': True})
```

**Pull request merged**